packages/ember-glimmer/tests/integration/components/closure-components-test.js

```diff
@@ -0,0 +1,669 @@
+
+    this.assertText(expectedText);
+
+    /*
+    this.runTask(() => this.context.set('expectedText', 'Hola'));
+
+    this.assertText('Hola');
+    */
+
+    this.runTask(() => this.context.set('expectedText', expectedText));
+  }
+
+  ['@test renders with dot path and with rest positional parameters']() {
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: 'params'
+      }),
+      template: '{{params}}'
+    });
+
+    let expectedText = 'Hodi';
+
+    this.render(strip`
+      {{#with (hash lookedup=(component "-looked-up")) as |object|}}
+        {{object.lookedup expectedText "Hola"}}
+      {{/with}}`, {
+        expectedText
+      }
+    );
+
+    this.assertText(`${expectedText},Hola`);
+
+    this.runTask(() => this.rerender());
+
+    this.assertText(`${expectedText},Hola`);
+
+    this.runTask(() => this.context.set('expectedText', 'Kaixo'));
+
+    this.assertText('Kaixo,Hola');
+
+    this.runTask(() => this.context.set('expectedText', expectedText));
+
+    this.assertText(`${expectedText},Hola`);
+  }
+
+  ['@test renders with dot path and rest parameter does not leak'](assert) {
+    // In the original implementation, positional parameters were not handled
+    // correctly causing the first positional parameter to be the closure
+    // component itself.
+    let value = false;
+
+    this.registerComponent('my-component', {
+      ComponentClass: Component.extend({
+        didReceiveAttrs() {
+          value = this.getAttr('value');
+        }
+      }).reopenClass({
+        positionalParams: ['value']
+      })
+    });
+
+    this.render(strip`
+      {{#with (hash my-component=(component 'my-component')) as |c|}}
+        {{c.my-component}}
+      {{/with}}`
+    );
+
+    assert.ok(isEmpty(value), 'value is an empty parameter');
+  }
+
+  ['@test renders with dot path and updates attributes'](assert) {
+    this.registerComponent('my-nested-component', {
+      ComponentClass: Component.extend({
+        didReceiveAttrs() {
+          this.set('myProp', this.getAttr('my-parent-attr'));
+        }
+      }),
+      template: '<span id="nested-prop">{{myProp}}</span>'
+    });
+
+    this.registerComponent('my-component', {
+      template: '{{yield (hash my-nested-component=(component "my-nested-component" my-parent-attr=my-attr))}}'
+    });
+
+    this.registerComponent('my-action-component', {
+      ComponentClass: Component.extend({
+        actions: {
+          changeValue() { this.incrementProperty('myProp'); }
+        }
+      }),
+      template: strip`
+        {{#my-component my-attr=myProp as |api|}}
+          {{api.my-nested-component}}
+        {{/my-component}}
+        <br>
+        <button onclick={{action 'changeValue'}}>Change value</button>`
+    });
+
+    this.render('{{my-action-component myProp=myProp}}', {
+      myProp: 1
+    });
+
+    assert.equal(this.$('#nested-prop').text(), '1');
```

Comment:
That `this.$` is defined in [`abstract-test-case`](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/utils/abstract-test-case.js#L141-L143). If we were to move away from jQuery, we would just need to change things there.